### PR TITLE
Fs 3581 update round definition to cater optional or mandatory application feedback survey questions

### PR DESCRIPTION
**Pull request by RamuniN**
https://dluhcdigital.atlassian.net/browse/FS-3581
https://dluhcdigital.atlassian.net/browse/FS-3582

### Description
- Make feedback survey questions to optional on frontend when the config is set
- Add the optional text when the survey is optional to fill

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
- set the value `is_feedback_survey_optional` to True in fund-store db and complete an application, observe that application can be submitted without the survey

### Screenshots of UI changes (if applicable)
1) `is_feedback_survey_optional` = False (Requires survey to be filled to submit the application)
Picture1
2) `is_feedback_survey_optional` = True  (Survey  is optional, application can be submitted without completing the survey)
Picture2

### CODE REVIEW

File: app/helpers.py [hunk 1]
```diff
@@ -286,46 +286,54 @@ def get_round(
...
-        for page_number in ["1", "2", "3", "4"]
-    }
+    if feedback_survey_config.requires_survey:
+        # we grab the number from the last section i.e "6. Foobar" then increment it to get "7. " for feedback.
+        last_section_title = section_display_config[-1].title
+        number = None
+        if re.search(r"\b(\d+)\.", last_section_title):
+            number = (
+                int(re.search(r"\b(\d+)\.", last_section_title).group(1)) + 1
+            )
```
Comment (tferns) [inline app/helpers.py:303]: Was going to complain about this code then realised I wrote it, and this PR just changed indentation 😂 
```suggestion
        _match = re.search(r"\b(\d+)\.", last_section_title)
        number = int(_match.group(1)) + 1 if match else None
```

File: app/models/round.py [hunk 2]
```diff
@@ -4,6 +4,13 @@
 from dataclasses import dataclass
 
 
+@dataclass
+class FeedbackSurveyConfig:
+    requires_survey: bool = False
+    isSurveyOptional: bool = True
```
Comment (tferns) [inline app/models/round.py:10]: Same question on camelCase and conflicting variables

File: app/helpers.py (hunk 1 above)
Comment (RamuniN) [reply app/helpers.py:303]: Do you want me to commit the suggestion ?

Comment (tferns) [reply app/helpers.py:303]: Up to you!

File: app/models/round.py [hunk 3]
```diff
@@ -24,8 +43,16 @@ class Round:
...
     mark_as_complete_enabled: bool = False
+    feedback_survey_config: FeedbackSurveyConfig = None
+
+    def __post_init__(self):
+        if isinstance(self.feedback_survey_config, dict):
+            self.feedback_survey_config = FeedbackSurveyConfig.from_dict(
+                self.feedback_survey_config
+            )
+        elif self.feedback_survey_config is None:
+            self.feedback_survey_config = FeedbackSurveyConfig()
```
Comment (tferns) [inline app/models/round.py:55]: Do we need this? Couldn't we just have `= FeedbackSurveyConfig()` instead of `= None`?

**Review by tferns — APPROVED** [review]
lgtm!

File: app/models/round.py (hunk 3 above)
Comment (RamuniN) [reply app/models/round.py:55]: Made to use default values here, if it's None

**Pull request merged**